### PR TITLE
Respect Kernel Term Sharing flag when unfolding primitives in conversion

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

File: kernel/reduction.ml [hunk 1]
```diff
@@ -303,7 +303,8 @@ let unfold_ref_with_args infos tab fl v =
   | Primitive op when check_native_args op v ->
     let c = match fl with ConstKey c -> c | _ -> assert false in
     let rargs, a, nargs, v = get_native_args1 op c v in
-    Some (a, (Zupdate a::(Zprimitive(op,c,rargs,nargs)::v)))
+    let v = zupdate infos a (Zprimitive(op,c,rargs,nargs)::v) in
```
Comment (ppedrot) [inline kernel/reduction.ml:306]: I think you should rather export a smart constructor for `Zprimitive` that does both the update and the primitive.

Comment (SkySkimmer) [reply kernel/reduction.ml]: changed

**Comment by SkySkimmer** [comment]
This doesn't work because `zupdate` locks the term, so in the flex/flex case we recurse with a locked term on one side

**Comment by SkySkimmer** [comment]
Too annoying to solve

**Pull request closed without merging**